scaleway/resource_compute_instance_server_test.go

```diff
@@ -0,0 +1,181 @@
+				),
+			},
+			{
+				Config: testAccCheckScalewayComputeInstanceServerConfigBasic1("x86_64", "DEV1-S"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayComputeInstanceServerExists("scaleway_compute_instance_server.base"),
+					resource.TestCheckResourceAttr("scaleway_compute_instance_server.base", "type", "DEV1-S"),
+					resource.TestCheckResourceAttr("scaleway_compute_instance_server.base", "name", "test"),
+					resource.TestCheckResourceAttr("scaleway_compute_instance_server.base", "tags.0", "terraform-test"),
+					resource.TestCheckResourceAttr("scaleway_compute_instance_server.base", "tags.1", "scaleway_compute_instance_server"),
+					resource.TestCheckResourceAttr("scaleway_compute_instance_server.base", "tags.2", "basic1"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckScalewayComputeInstanceServerExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("resource not found: %s", n)
+		}
+
+		instanceApi, zone, ID, err := getInstanceAPIWithZoneAndID(testAccProvider.Meta(), rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		_, err = instanceApi.GetServer(&instance.GetServerRequest{ServerID: ID, Zone: zone})
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckScalewayComputeInstanceServerDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "scaleway_compute_instance_ip" {
+			continue
+		}
+
+		instanceApi, zone, ID, err := getInstanceAPIWithZoneAndID(testAccProvider.Meta(), rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		_, err = instanceApi.GetServer(&instance.GetServerRequest{
+			ServerID: ID,
+			Zone:     zone,
+		})
+
+		// If no error resource still exist
+		if err == nil {
+			return fmt.Errorf("Server (%s) still exists", rs.Primary.ID)
+		}
+
+		// Unexpected api error we return it
+		if !is404Error(err) {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckScalewayComputeInstanceServerConfigMinimal1() string {
+	return `
+resource "scaleway_compute_instance_server" "webserver" {
+image = "f974feac-abae-4365-b988-8ec7d1cec10d"
+type  = "DEV1-S"
+tags = [ "terraform-test", "scaleway_compute_instance_server", "minimal1" ]
+}`
+}
+
+func testAccCheckScalewayComputeInstanceServerConfigBasic1(architecture, serverType string) string {
+	return fmt.Sprintf(`
+data "scaleway_image" "ubuntu" {
+  architecture = "%s"
+  name         = "Ubuntu Xenial"
+  most_recent  = true
+}
+
+resource "scaleway_compute_instance_server" "base" {
+  name = "test"
+  image = "${data.scaleway_image.ubuntu.id}"
+  type = "%s"
+  tags = [ "terraform-test", "scaleway_compute_instance_server", "basic1" ]
+}`, architecture, serverType)
+}
+
+// todo: add tests with IP attachment
+
+// todo: add tests with additional volume attachement
```

Comment:
No need This will be tested by volume

Reply:
That's not really a functionally we have on `scaleway_compute_instance_volume` resource. Or are you talking about creating a volume attachment resource?